client/gribiclient.go

```diff
@@ -201,8 +208,20 @@ func (c *Client) UseStub(stub spb.GRIBIClient) error {
 	return nil
 }
 
+// disconnect shuts down the goroutines started by Connect().
+func (c *Client) disconnect() {
+	if !c.started.Load() || c.shut.Load() {
+		// goroutines have not started or have already exited, so c.q()
+		// would deadlock.
+		return
+	}
+	c.q(nil) // signals reqHandler to close the stream.
```

Comment:
I'm not sure if using `nil` is the best way to signal closing. 

Shall we somehow take advantage of `isConverged()` and directly call `stream.CloseSend()` here? Is that a good idea?